test/connector/tcp/mssql/mssql_client_request.go

```diff
@@ -0,0 +1,162 @@
+func cloneCredentials(original map[string][]byte) map[string][]byte {
+	credsClone := make(map[string][]byte)
+
+	for key, value := range original {
+		// Clone the value
+		valueClone := make([]byte, len(value))
+		copy(valueClone, value)
+
+		// Set the key, value pair on the credentials clone
+		credsClone[key] = valueClone
+	}
+
+	return credsClone
+}
+
+// proxyRequest issues a client request using a the 'executor' argument to a Secretless
+// proxy service configured using the 'credentials' argument.
+// proxyRequest uses newInProcessProxyService to creating the in-process proxy service.
+// The proxy service exists only for the lifetime of this method call.
+func (clientReq clientRequest) proxyRequest(
+	executor dbClientExecutor,
+	credentials map[string][]byte,
+) (string, string, error) {
+	// Create in-process proxy service
+	proxyService, err := newInProcessProxyService(credentials)
+	if err != nil {
+		return "", "", err
+	}
+
+	// Ensure the proxy service is stopped
+	defer proxyService.Stop()
+	// Start the proxyService service
+	proxyService.Start()
+
+	// Make the client request to the proxy service
+	clientResChan := concurrentClientExec(
+		executor,
+		dbClientConfig{
+			Host:     proxyService.host,
+			Port:     proxyService.port,
+			Username: "dummy",
+			Password: "dummy",
+			Database: clientReq.database,
+			ReadOnly: clientReq.readOnly,
+		},
+		clientReq.query,
+	)
+
+	// Block and wait for the client response
+	clientRes := <-clientResChan
+
+	return clientRes.out, proxyService.port, clientRes.err
+}
+
+// proxyToCreatedMock issues a client request using a the 'executor' argument to a Secretless
+// proxy service configured using the 'credentials' argument.
+// proxyRequest uses newInProcessProxyService to creating the in-process proxy service.
+// The proxy service exists only for the lifetime of this method call.
+//
+// NOTE: proxyToCreatedMock proxies the request to a mock server that terminates the request after the handshake
+// This can have unintended effects. gomssql in particular does some weird retry, when a query is prepared!
+// TODO: find out this weird gomssqldb behavior.
+func (clientReq clientRequest) proxyToCreatedMock(
+	executor dbClientExecutor,
+	credentials map[string][]byte,
+) (*mockTargetCapture, string, error) {
+	// Create mock target
+	mt, err := newMockTarget("0")
+	if err != nil {
+		return nil, "", err
+	}
+	defer mt.close()
+
+	// Gather credentials
+	baseCredentials := map[string][]byte{
+		"host": []byte(mt.host),
+		"port": []byte(mt.port),
+	}
+	for key, value := range credentials {
+		baseCredentials[key] = value
+	}
+
+	// Accept on mock target
+	mtResChan := mt.singleAcceptAndHandle()
+
+	// We don't expect anything useful to come back from the client request.
+	// This is a fire and forget
+	_, secretlessPort, _ := clientReq.proxyRequest(
+		executor,
+		baseCredentials,
+	)
+
+	mtRes := <-mtResChan
```

Comment:
Not a blocker but this chan op and the one in function below should have a comment that they're blocking the flow while waiting so that it's very obvious that the flow might get stuck